StereoKit/Math/Ray.cs

```diff
@@ -165,6 +166,16 @@ public bool Intersect(Model model, out Ray modelSpaceAt)
 		public bool Intersect(Model model, Cull cullFaces, out Ray modelSpaceAt)
 			=> NativeAPI.model_ray_intersect(model._inst, this, cullFaces, out modelSpaceAt);
 
+		/// <inheritdoc cref="Intersect(Model, Cull, out Ray)"/>
+		/// <param name="modelNode">ModelNode that intersects with ray.</param>
+		public bool Intersect(Model model, Cull cullFaces, out Ray modelSpaceAt, out ModelNode modelNode)
+		{
+			int modelNodeId = -1;
```

Comment:
C#'s rule for `out` parameters is that the function _must_ give the `out` parameter _a_ value. While C++ does not have this rule, this is a soft rule I follow when writing C++ code as well. So, this initialization can be skipped and delegated to `mode_ray_intersect`, which allows us to create the variable inline: `model_ray_intersect(model._inst, this, cullFaces, out modelSpaceAt, out int modelNodeId)`